data_structures/static_rtree.hpp

```diff
@@ -567,589 +342,130 @@ class StaticRTree
-                                m_coordinate_list->at(current_segment.v));
-
-                const float backward_edge_bearing = (forward_edge_bearing + 180) > 360 
-                                                      ? (forward_edge_bearing - 180) 
-                                                      : (forward_edge_bearing + 180);
-
-                const bool forward_bearing_valid = IsBearingWithinBounds(forward_edge_bearing, filter_bearing, filter_bearing_range)
-                    && current_segment.forward_edge_based_node_id != SPECIAL_NODEID;
-                const bool backward_bearing_valid = IsBearingWithinBounds(backward_edge_bearing, filter_bearing, filter_bearing_range)
-                    && current_segment.reverse_edge_based_node_id != SPECIAL_NODEID;
-
-                if (!forward_bearing_valid && !backward_bearing_valid)
-                {
-                    // This edge doesn't fall within our bearing filter
                     continue;
                 }
 
                 // store phantom node in result vector
-                result_phantom_node_vector.emplace_back(
-                    PhantomNode {current_segment, foot_point_coordinate_on_segment},
-                    current_perpendicular_distance);
-
-                if (!forward_bearing_valid)
-                {
-                    result_phantom_node_vector.back().first.forward_node_id = SPECIAL_NODEID;
-                }
-                if (!backward_bearing_valid)
-                {
-                    result_phantom_node_vector.back().first.reverse_node_id = SPECIAL_NODEID;
-                }
-
-                // Hack to fix rounding errors and wandering via nodes.
-                FixUpRoundingIssue(input_coordinate, result_phantom_node_vector.back().first);
-
-                // set forward and reverse weights on the phantom node
-                SetForwardAndReverseWeightsOnPhantomNode(current_segment,
-                                                         result_phantom_node_vector.back().first);
-            }
-
-            // stop the search by flushing the queue
-            if (inspected_elements >= max_checked_elements)
-            {
-                traversal_queue = std::priority_queue<IncrementalQueryCandidate>{};
-            }
-        }
-
-        return !result_phantom_node_vector.empty();
-    }
-
-    bool FindPhantomNodeForCoordinate(const FixedPointCoordinate &input_coordinate,
-                                      PhantomNode &result_phantom_node,
-                                      const unsigned zoom_level)
-    {
-        const bool ignore_tiny_components = (zoom_level <= 14);
-        EdgeDataT nearest_edge;
-
-        float min_dist = std::numeric_limits<float>::max();
-        float min_max_dist = std::numeric_limits<float>::max();
-
-        std::priority_queue<QueryCandidate> traversal_queue;
-        traversal_queue.emplace(0.f, 0);
+                results.emplace_back(std::move(current_segment));
```

Comment:
When the container stores Ts and you want to move a T into it use the `push_back(&&)` overload and move it in.
When the container stores Ts that can be constructed from U, V (has a constructor T(U, V)) and you want to move a U or V into it, use `emplace_back(Args&&...)` and move the U and V arguments in.